lib/url.c

```diff
@@ -3485,8 +3486,7 @@ static void reuse_conn(struct Curl_easy *data,
 
   /* get the user+password information from the old_conn struct since it may
    * be new for this request even when we re-use an existing connection */
-  conn->bits.user_passwd = old_conn->bits.user_passwd;
-  if(conn->bits.user_passwd) {
+  if(old_conn->user) {
```

Comment:
the other checks that replaced user_passwd are now using the allocated per data aptr.user !=NULL but this one is using conn->user (and became old_conn->user) so now if conn->user and aptr.user are out of sync wouldn't this be as well, in other words you're copying the user/passwd even if aptr.user was set to NULL, unless they are both kept in sync (see earlier comment)